src/_pytest/fixtures.py

```diff
@@ -226,17 +227,19 @@ def reorder_items(items: Sequence[nodes.Item]) -> List[nodes.Item]:
 def fix_cache_order(
     item: nodes.Item,
     argkeys_cache: Dict[Scope, Dict[nodes.Item, Dict[FixtureArgKey, None]]],
-    items_by_argkey: Dict[Scope, Dict[FixtureArgKey, "Deque[nodes.Item]"]],
+    items_by_argkey: Dict[Scope, Dict[FixtureArgKey, OrderedDict[nodes.Item, None]]],
 ) -> None:
     for scope in HIGH_SCOPES:
+        scoped_items_by_argkey = items_by_argkey[scope]
         for key in argkeys_cache[scope].get(item, []):
-            items_by_argkey[scope][key].appendleft(item)
+            scoped_items_by_argkey[key][item] = None
+            scoped_items_by_argkey[key].move_to_end(item, last=False)
```

Comment:
Oh.. So this was backported after all. Makes sense given that the regression appeared in v8.2.2.